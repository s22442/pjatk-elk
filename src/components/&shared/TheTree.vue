<script setup lang="ts">
const { ready, start } = $(
  useTimeout(1000, { controls: true, immediate: false })
);

const shake = $computed(() => !ready);

const handleClick = () => {
  if (ready) {
    start();
  }
};
</script>

<template>
  <div
    :class="{ shake }"
    _icon-carbon-tree
    _text="secondary hover:accent"
    _transition-colors
    _cursor-pointer
    @click="handleClick()"
  />
</template>

<style lang="scss" scoped>
.shake {
  animation: shake 0.8s cubic-bezier(0.36, 0.07, 0.19, 0.97) both;
  transform: translate3d(0, 0, 0);
}

@keyframes shake {
  10%,
  90% {
    transform: translate3d(-1px, 0, 0) rotate(-1deg);
  }

  20%,
  80% {
    transform: translate3d(2px, 0, 0) rotate(2deg);
  }

  30%,
  50%,
  70% {
    transform: translate3d(-4px, 0, 0) rotate(-4deg);
  }

  40%,
  60% {
    transform: translate3d(4px, 0, 0) rotate(4deg);
  }
}
</style>
