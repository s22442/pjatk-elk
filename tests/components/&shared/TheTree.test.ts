import TheTree from '~/components/&shared/TheTree.vue';

import { mount } from '@vue/test-utils';

describe('TheTree', () => {
  it('should render', () => {
    const wrapper = mount(TheTree);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should shake', async () => {
    const wrapper = mount(TheTree);
    await wrapper.trigger('click');
    expect(wrapper.classes('shake')).toBe(true);
  });
});
