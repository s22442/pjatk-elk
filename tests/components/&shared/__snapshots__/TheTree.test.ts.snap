// Vitest Snapshot v1

exports[`TheTree > should render 1`] = `"<div class=\\"\\" _icon-carbon-tree=\\"\\" _text=\\"secondary hover:accent\\" _transition-colors=\\"\\" _cursor-pointer=\\"\\" data-v-84dc2214=\\"\\"></div>"`;
